internal/controller/driver_controller.go

```diff
@@ -967,6 +995,8 @@ func (r *driverReconcile) reconcileNodePluginDeamonSet() error {
 							utils.PluginMountDirVolume(pluginSpec.KubeletDirPath),
 							utils.PodsMountDirVolume(pluginSpec.KubeletDirPath),
 							utils.RegistrationDirVolume(pluginSpec.KubeletDirPath),
+							utils.If(r.logRotation(), utils.LogDirVolume(r.driver.Spec.Log.Rotation.LogHostPath, daemonSet.Name), corev1.Volume{}),
+							utils.If(r.logRotation(), utils.LogSidecarDirVolume(r.driver.Spec.Log.Rotation.LogHostPath, daemonSet.Name), corev1.Volume{}),
```

Comment:
You should not use `utils.If` and an empty volume definition in the volume list.
See a previous comment for the reasoning